build/commands/scripts/commands.js

```diff
@@ -254,25 +254,50 @@ program
     build('Release', options)
   })
 
-program
-  .command('test <suite>')
-  .allowUnknownOption(true)
-  .option('-C <build_dir>', 'build config (out/Debug, out/Release')
-  .option('--v [log_level]', 'set log level to [log_level]', parseInteger, '0')
-  .option('--vmodule [modules]', 'verbose log from specific modules')
-  .option('--filter <filter>', 'set test filter')
-  .option('--output <output>', 'set test output (results) file path')
-  .option('--disable_brave_extension', 'disable loading the Brave extension')
-  .option('--single_process', 'uses a single process to run tests to help with debugging')
-  .option('--test_launcher_jobs <test_launcher_jobs>', 'Number of jobs to launch', parseInteger, '4')
-  .option('--target_os <target_os>', 'target OS')
-  .option('--target_arch <target_arch>', 'target architecture')
-  .option('--run_disabled_tests', 'run disabled tests')
-  .option('--manual_android_test_device', 'indicates that Android test device is run manually')
-  .option('--use_goma [arg]', 'whether to use Goma for building', JSON.parse)
-  .option('--goma_offline', 'use offline mode for goma')
-  .arguments('[build_config]')
-  .action(test.bind(null, parsedArgs.unknown))
+// We have a single `test` command for developer convenience. It consists of
+// two steps: Building and running. However, if it fails, then it is unclear
+// which of the two steps failed. This is a problem in CI, where we don't
+// want to produce test reports when building fails. So we also have separate
+// `build_tests` and `run_tests` commands.
+
+const testCommand = program.command('test <suite>');
+const buildTestsCommand = program.command('build_tests <suite>');
+const runTestsCommand = program.command('run_tests <suite>');
+
+[testCommand, buildTestsCommand, runTestsCommand].forEach((command) => {
```

Comment:
I think this would be a lot simpler if we had an optional `--target` param for the `build` command. Separately we should do the same thing for create_dist

Reply:
for backwards compat you change package.json for `npm run test` to handle this as two commands

Reply:
> I think this would be a lot simpler if we had an optional --target param for the build

We have such a target, don't we? For [example](https://github.com/brave/brave-core/pull/15440):

    npm run build -- --target=brave/components/policy:pack_policy_templates

I'm pretty sure I'm not fully understanding what you wrote. Could you clarify in more detail what you meant?

PS - as always ty for the review.

Reply:
> for backwards compat you change package.json for npm run test to handle this as two commands

I'm afraid I don't understand the message of this comment either. Yes I'm changing package.json in a way that's backwards compatible (while also retaining the `test` command for developer ease). Is there something I should change?

Reply:
@bridiver could I ask you to clarify your comments?